wideq/dryer.py

```diff
@@ -0,0 +1,205 @@
+    SIXTY = '60'
+    UNKNOWN = _UNKNOWN
+
+
+class DryerDevice(Device):
+    """A higher-level interface for a dryer."""
+
+    def poll(self) -> Optional['DryerDevice']:
+        """Poll the device's current state.
+
+        Monitoring must be started first with `monitor_start`.
+
+        :returns: Either a `DryerStatus` instance or `None` if the status is
+            not yet available.
+        """
+        # Abort if monitoring has not started yet.
+        if not hasattr(self, 'mon'):
+            return None
+
+        res = self.mon.poll_json()
+        if res:
+            return DryerStatus(self, res)
+        else:
+            return None
+
+
+class DryerStatus(object):
+    """Higher-level information about a dryer's current status.
+
+    :param dryer: The DryerDevice instance.
+    :param data: JSON data from the API.
+    """
+
+    def __init__(self, dryer: DryerDevice, data: dict):
+        self.dryer = dryer
+        self.data = data
+
+    def get_bit(self, key: str, index: int) -> str:
+        bit_value = int(self.data[key])
+        bit_index = 2 ** index
+        mode = bin(bit_value & bit_index)
+        if mode == bin(0):
+            return 'OFF'
+        else:
+            return 'ON'
+
+    @property
+    def state(self) -> DryerState:
+        """Get the state of the dryer."""
+        attr = 'State'
+        return DryerState(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def previous_state(self) -> DryerState:
+        """Get the previous state of the dryer."""
+        attr = 'PreState'
+        return DryerState(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def dry_level(self) -> DryLevel:
+        """Get the dry level."""
+        attr = 'DryLevel'
+        return DryLevel(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def temperature_control(self) -> TempControl:
+        """Get the temperature control setting."""
+        attr = 'TempControl'
+        return TempControl(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def time_dry(self) -> TimeDry:
+        """Get the time dry setting."""
+        attr = 'TimeDry'
+        return TimeDry(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def is_on(self) -> bool:
+        """Check if the dryer is on or not."""
+        return self.state != DryerState.OFF
+
+    @property
+    def remain_time_hours(self):
+        """Get the remaining number of hours."""
+        return self.data['Remain_Time_H']
+
+    @property
+    def remain_time_minutes(self):
+        """Get the remaining number of minutes."""
+        return self.data['Remain_Time_M']
+
+    @property
+    def initial_time_hours(self):
+        """Get the initial number of hours."""
+        return self.data['Initial_Time_H']
+
+    @property
+    def initial_time_minutes(self):
+        """Get the initial number of minutes."""
+        return self.data['Initial_Time_M']
```

Comment:
For these values, does the API report a number of hours and minutes separately? For example, if `Initial_Time_H` is 1 and `Initial_Time_M` is 30, does that mean that the indicated time is 90 minutes?

If so, that strikes me as a pretty silly design. :smiley: Maybe it would be more convenient for clients if we were to wrap this up as just a number of minutes—i.e., the `initial_time` property would just contain 90.

Reply:
Unfortunately yes, your example is actually how the time is reported (90 minutes would show the initinal_time_h as 1 and initial_time_m as 30).  

I agree with your suggestion to just use a single attribute and return the time in minutes.